python/src/aiconfig/default_parsers/azure.py

```diff
@@ -0,0 +1,40 @@
+from aiconfig.default_parsers.openai import DefaultOpenAIParser
+from aiconfig.util.config_utils import get_api_key_from_environment
+
+from openai import AzureOpenAI
+
+
+class AzureParser(DefaultOpenAIParser):
+    def __init__(self, deployment: str):
+        super().__init__(model_id = deployment)
+
+        # Deployment is a custom name that user defines in Azure portal. 
+        self.deployment = deployment 
+
+    async def deserialize(self, *args, **kwargs):
+        # Logic doesn't depend on input, pass it forward to the openai model parser deserialize 
+        openai_deserialized_params = await super().deserialize(*args, **kwargs)
+
+        # Should this just be the "model name" from config? 
+        # If so, Then the "model name" will have to be deployment name and be consistent 
+        # Which would mean we don't need to have this custom deserialization
+        openai_deserialized_params["model"] = self.deployment 
+        return openai_deserialized_params
+
+    def initialize_openai_client(self) -> None:
+        ## The Azure Client itself will check and retrieve these from environment 
+        # if its not explicitly passed in & will raise otherwise. Technically not needed
+        azure_openai_key = get_api_key_from_environment(
+            "AZURE_OPENAI_KEY"
+        ).unwrap()
+
+        azure_endpoint = get_api_key_from_environment(
```

Comment:
Is there a default in case this isn't set? Or does this have to be set?

Reply:
Nope there's no defaults. It has to be set otherwise it will throw an exception if one of the 3 required ones are missing

Reply:
I think our error messages are a lot easier to understand but its the same error being thrown anyway